moveit_ros/planning_interface/moveit_cpp/include/moveit/moveit_cpp/moveit_cpp.h

```diff
@@ -0,0 +1,1099 @@
+      CurrentStateMonitor instance is automatically constructed.  This
+      function allows triggering the construction of that object from
+      the beginning, so that future calls to functions such as
+      getCurrentState() will not take so long and are less likely to fail. */
+  bool startStateMonitor(double wait = 1.0);
+
+  /** \brief Get the current joint values for the joints planned for by this instance (see getJoints()) */
+  std::vector<double> getCurrentJointValues();
+
+  /** \brief Get the current state of the robot within the duration specified by wait. */
+  robot_state::RobotStatePtr getCurrentState(double wait = 1);
+
+  /** \brief Get the pose for the end-effector \e end_effector_link.
+      If \e end_effector_link is empty (the default value) then the end-effector reported by getEndEffectorLink() is
+     assumed */
+  geometry_msgs::PoseStamped getCurrentPose(const std::string& end_effector_link = "");
+
+  /** \brief Get the roll-pitch-yaw (XYZ) for the end-effector \e end_effector_link.
+      If \e end_effector_link is empty (the default value) then the end-effector reported by getEndEffectorLink() is
+     assumed */
+  std::vector<double> getCurrentRPY(const std::string& end_effector_link = "");
+
+  /** \brief Get random joint values for the joints planned for by this instance (see getJoints()) */
+  std::vector<double> getRandomJointValues();
+
+  /** \brief Get a random reachable pose for the end-effector \e end_effector_link.
+      If \e end_effector_link is empty (the default value) then the end-effector reported by getEndEffectorLink() is
+     assumed */
+  geometry_msgs::PoseStamped getRandomPose(const std::string& end_effector_link = "");
+
+  /**@}*/
+
+  /**
+   * \name Manage named joint configurations
+   */
+  /**@{*/
+
+  /** \brief Remember the current joint values (of the robot being monitored) under \e name.
+      These can be used by setNamedTarget().
+      These values are remembered locally in the client.  Other clients will
+      not have access to them. */
+  void rememberJointValues(const std::string& name);
+
+  /** \brief Remember the specified joint values  under \e name.
+      These can be used by setNamedTarget().
+      These values are remembered locally in the client.  Other clients will
+      not have access to them. */
+  void rememberJointValues(const std::string& name, const std::vector<double>& values);
+
+  /** \brief Get the currently remembered map of names to joint values. */
+  const std::map<std::string, std::vector<double> >& getRememberedJointValues() const
+  {
+    return remembered_joint_values_;
+  }
+
+  /** \brief Forget the joint values remebered under \e name */
+  void forgetJointValues(const std::string& name);
+
+  /**@}*/
+
+  /**
+   * \name Manage planning constraints
+   */
+  /**@{*/
+
+  /** \brief Specify where the database server that holds known constraints resides */
+  void setConstraintsDatabase(const std::string& host, unsigned int port);
```

Comment:
@mlautman @henningkayser have you ever seen this feature used? I don't understand what it is. I would vote removing all this stuff until we see a need for it in the future.

Reply:
Never used it, I'll remove it